### PR TITLE
Add calico as an alternative CNI plugin

### PR DESCRIPTION
**Pull request by nupplaphil**
Because I want to use Istio with the cluster, based on this repo, I now added the possibility to use calico too.

I tried at least these connection tests:
- https://projectcalico.docs.tigera.io/security/tutorials/kubernetes-policy-basic
- https://projectcalico.docs.tigera.io/security/tutorials/kubernetes-policy-advanced

And all steps went fine without any error

Based on the suggestions at #137

### CODE REVIEW

**Comment by mysticaltech** [comment]
Hey @nupplaphil, thank you for this PR! Looking forward to the few tweaks above. and being able to merge them.

**Comment by nupplaphil** [comment]
@mysticaltech - from my point of view, the PR was done already, do you see any troubles merging it?
tbh I didn't thought about adding cilium as well because I don't need it (yet).

**Comment by mysticaltech** [comment]
Cilium is not the problem here @nupplaphil. There are issues in this PR, see my comments above. When these are addressed then definitely I will be happy to merge.

**Comment by nupplaphil** [comment]
tbh - did you really send your review? I don't see any comments

**Review by mysticaltech — CHANGES_REQUESTED** [review]

**Comment by mysticaltech** [comment]
> tbh - did you really send your review? I don't see any comments

Apologies, indeed I had not submitted my reviews!  You guessed right 🙏

File: control_planes.tf [hunk 1]
```diff
@@ -43,19 +43,26 @@ resource "null_resource" "control_planes" {
...
-      flannel-iface            = "eth1"
       kubelet-arg              = "cloud-provider=external"
       node-ip                  = module.control_planes[count.index].private_ipv4_address
       advertise-address        = module.control_planes[count.index].private_ipv4_address
       node-taint               = var.allow_scheduling_on_control_plane ? [] : ["node-role.kubernetes.io/master:NoSchedule"]
       node-label               = var.automatically_upgrade_k3s ? ["k3s_upgrade=true"] : []
-    })
+      },
+      var.cni_plugin == "flannel" ? {
+        flannel-iface = "eth1"
```
[the remark this answers is not in the document]
Comment (mysticaltech) [reply control_planes.tf:59]: Here talking about line 59.

**Comment by nupplaphil** [comment]
OK, back to draft, I had a strange flexvolume error, I'll investigate it first

**Comment by mysticaltech** [comment]
Don't forget to do the changes I suggested. Could solve your problem, especially the eth1.

**Comment by nupplaphil** [comment]
I already changed it, but a pod of a test nginx throws a strange Felix driver (uds) not found exception. I guess I have to check the paths :)

But thanks for the suggestion :)

**Comment by nupplaphil** [comment]
Now it works.

We now set the "volume-plugin-dir" to `/var/lib/kubelet/volumeplugins` independent if we use calico, because the default directory would be set to `/usr/libexec/kubernetes/kubelet-plugins/volume/exec/`, which is wrong for the whole setup (=> `/usr` is readonly). 

It's now working for istio as well (successfully set mTLS per calico & cert-manager for pods inside the istio mesh)

**Comment by mysticaltech** [comment]
Amazing, good find @nupplaphil!

**Review by mysticaltech — APPROVED** [review]

**Comment by mysticaltech** [comment]
I will just run one deploy, ASAP this week, and if all checks out, will merge.

**Comment by mysticaltech** [comment]
Hey @nupplaphil, it would be an honor for us to have you join Kube Hetzner (I have just invited you on Github). If you accept, you will become a member of our team and have access to branches, issues... etc. And help steer the future of this project.

And here's an invitation to our private chat room on Matrix (I use the element.io client) [nupplaphil](https://matrix.to/#/#kube-hetzner:matrix.org) (will delete later)! :)

**Comment by nupplaphil** [comment]
Hi  @mysticaltech , thanks for your invitation! I'll happily accept it, but I'm not sure, how much time I can offer to support this project. I'll do my best :-)

I'm currently setting up my Matrix homeserver and will join the group as soon as possible

**Comment by mysticaltech** [comment]
Welcome, welcome! 🙏 Don't worry about time if you see the others; some do not show for weeks! 😂

All are just voluntary contributions; we're like a distributed system! When someone has an idea to move the project forward, they chip in. Nothing more!

Very happy that you are deploying your Matrix server! Last time, matrix.org was not working amazingly, probably it was overloaded or something. So great idea! See you on the other side! :) 

Please do copy the link, to join above, then give a thumb up, and I will delete all these messages as not related to the PR.

**Pull request merged**